src/decoder/mod.rs

```diff
@@ -635,6 +641,21 @@ impl<R: BufRead + Seek> Reader<R> {
         size * height as usize
     }
 
+    /// Return the number of bytes required to hold a deinterlaced image frame, if it fits into the
+    /// memory space of the machine.
+    pub fn checked_output_buffer_size(&self) -> Option<usize> {
+        let (width, height) = self.info().size();
+        let (color, depth) = self.output_color_type();
+        // The subtraction should always work, but we do this for consistency. Also note that by
+        // calling `checked_raw_row_length` the row buffer is guaranteed to work whereas if we
+        // ran other function that didn't include the filter byte that could later fail on an image
+        // that is `1xN`...
+        let size = color.checked_raw_row_length(depth, width)?.checked_sub(1)?;
+        let height = usize::try_from(height).ok()?;
+        size.checked_mul(height)
+            .filter(|&n| n < isize::MAX as usize)
```

Comment:
I find this line really confusing. Could you express it as an `if let` or a `match` for clarity?

Reply:
I opted for `bool::then`, is that alright to read?

Reply:
Yeah that's clear enough. Thanks!